src/puffin/src/file_format/writer/footer.rs

```diff
@@ -70,10 +77,13 @@ impl FooterWriter {
     }
 
     /// Appends reserved flags (currently zero-initialized) to the given buffer.
-    ///
-    /// TODO(zhongzc): support compression
     fn write_flags(&self, buf: &mut Vec<u8>) {
-        buf.extend_from_slice(&Flags::DEFAULT.bits().to_le_bytes());
+        let mut flags = Flags::DEFAULT;
+        if self.lz4_compressed {
```

Comment:
Also explicitly clear the bit field is `self.lz4_compressed` is false:
```suggestion
        if self.lz4_compressed {
            flags |= Flags::FOOTER_PAYLOAD_COMPRESSED_LZ4;
        } else {
            flags &= !Flags::FOOTER_PAYLOAD_COMPRESSED_LZ4;
        }
```